plugins/contract-sizes.ts

```diff
@@ -0,0 +1,575 @@
+import {Table} from 'console-table-printer'
+import {extendConfig, task, types} from 'hardhat/config'
+import {
+    HardhatConfig,
+    HardhatRuntimeEnvironment,
+    SolcConfig
+} from 'hardhat/types'
+import chalk from 'chalk'
+
+import fs from 'fs'
+import path from 'path'
+
+const MAX_CONTRACT_SIZE = 24576 // applied to color contract size as green / yellow / red
```

Comment:
suggestion: have this as file or env config with this as the default when absent.

Ideal contract sizing changes across networks. Specifically I'm thinking of the layer 2, that have support for much larger contract sizes.

Reply:
this command is not related to any network, so
- filtering limits are provided via arguments
- and this one is only for coloring / cosmetics

ok, will make it as a CLI argument - making it as a per-network extension of hardhat config is an overkill